src/transport_ofi.c

```diff
@@ -563,36 +615,71 @@ static inline int atomic_limitations_check(void)
     }
 
     /* Binary OPS check */
-    for(i=0; i<3; i++) {//DT
-      for(j=0; j<3; j++) { //OPS
+    for(i=0; i<SIZEOF_DT_INT; i++) {//DT
+      for(j=0; j<SIZEOF_BOPS; j++) { //OPS
         ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_INT[i], SHM_BOPS[j],
                         &atomic_size);
-        if(ret!=0 || atomic_size == 0) {
-           OFI_ERRMSG("ret=%d atomic_size=%d %d %d\n", ret, (int)atomic_size, i, j);
-	   return ret;
-        }
+         if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_BOPS[j], SHM_NAMED_DT_INT[i]))
+           return ret;
       }
     }
 
     /* OTHER OPS check */
-    for(i=0; i<5; i++) {//DT
+    for(i=0; i<SIZEOF_DT_CMP; i++) {//DT
       for(j=0; j<4; j++) { //OPS
         ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_CMP[i], SHM_OPS[j],
                         &atomic_size);
-        if(ret!=0 || atomic_size == 0) {
-           OFI_ERRMSG("ret=%d atomic_size=%d %d %d\n", ret, (int)atomic_size, i, j);
-	   return ret;
-        }
+         if(atomicvalid_rtncheck(ret, atomic_size, atomicwarn,
+                            SHM_NAMED_OPS[j], SHM_NAMED_DT_CMP[i]))
+           return ret;
+      }
+    }
+
+    for(i=0; i<3; i++) {//DT does not include short
```

Comment:
Pull out this constant, too, please.  I would prefer a separate constant (rather than arithmetic on the existing constant) that is well documented to capture the convention in how the array is ordered.